Client/Frontend/Home/BookmarksPanel.swift

```diff
@@ -135,7 +135,7 @@ class BookmarksPanel: SiteTableViewController, HomePanel {
     }
 
     private func updateEmptyPanelState() {
-        if source?.current.count == 0 {
+        if source?.current.count == 0 && parentFolders.isEmpty {
```

Comment:
I would prefer `&& source?.current.guid == BookmarkRoots.MobileFolderGUID`. I don't trust the parent folder stack to always be correct and to be an invariant.